src/main/java/io/cos/cas/osf/web/flow/login/OsfDefaultLoginPreparationAction.java

```diff
@@ -74,15 +75,17 @@ protected Event doExecute(RequestContext context) {
                     encodedServiceUrl,
                     institutionLogin,
                     institutionId,
+                    unsupportedInstitutionLogin,
                     orcidRedirect,
                     orcidLoginUrl,
                     defaultService,
                     defaultServiceUrl
             );
         } else {
-            loginContext.setDefaultServiceUrl(encodedServiceUrl);
+            loginContext.setEncodedServiceUrl(encodedServiceUrl);
```

Comment:
cc @felliott , after searching our chat history, I found the issue you mentioned when working on SSO a while ago

Reply:
🎉  Chat archaeology FTW!